docs/DesignDocs/InitTimeConfigurationInterface.md

```diff
@@ -83,28 +83,66 @@ typedef enum _oe_enclave_setting_type
      *  reflected in TEE-produced enclave identity evidence. Currently only
      *  supported by SGX Enclaves with KSS feature enabled.
      */
-    OE_SGX_CONFIGURATION_DATA = 0x78b5b41d
+    OE_SGX_ENCLAVE_CONFIG_DATA = 0x78b5b41d
 } oe_enclave_setting_type_t;
 
-typedef struct _oe_sgx_configuration_data
+typedef struct _oe_sgx_enclave_setting_config_data
 {
-    uint8_t config_id[8];
+    uint8_t config_id[64];
     uint16_t config_svn;
+    bool ignore_if_unsupported;
+
 }
-oe_sgx_configuration_data_t;
+oe_sgx_enclave_setting_config_data;
 ```
 
-For SGX, based on whether `OE_SGX_CONFIGURATION_DATA` is provided and whether
+For SGX, based on whether `OE_SGX_ENCLAVE_CONFIG_DATA` is provided and whether
 SGX-KSS feature is supported, the SGX enclave loader sets `SECS.CONFIGID` and
-`SECS.CONFIGSVN` according to the table below.
-
-| CONFIGURATION_DATA | Behavior
-|-------|-----------------------------------
-|   -   | On systems where SGX-KSS feature is not available or disabled: No Action; On systems with SGX-KSS enabled: loader sets `SECS.CONFIGID` and `SECS.CONFIGSVN` as 0
-|   x   | On systems where SGX-KSS feature is not available or disabled: Invalid;  On systems with SGX-KSS enabled: loader copies _oe_sgx_configuration_data to `SECS.CONFIGID` and `SECS.CONFIGSVN`
+`SECS.CONFIGSVN` according to the table below. Developer can set the `ignore_if_unsupported`
+field to `true` to ignore the provided configuration data if the SGX-KSS feature is
+not supported.
+
+| SGX KSS Feature | CONFIGURATION_DATA | ignore_if_unsupported | Behavior
+|-------|------|----------|----------------
+| Supported | Provided   | true/false | loader sets `SECS.CONFIGID` and `SECS.CONFIGSVN` from oe_sgx_enclave_setting_config_data
+| Supported | Not Provided   | true/false | `SECS.CONFIGID` and `SECS.CONFIGSVN` are not changed
+| Not supported | Provided   | true | Enclave creation fails
+| Not supported | Provided   | false | Enclave loader ignores provided oe_sgx_enclave_setting_config_data
+| Not supported | Not Provided   | true/false | `SECS.CONFIGID` and `SECS.CONFIGSVN` are not changed
 
 The enclave developer is responsible for the host side code that produces the
-Enclave Configuration Data and pass the data to the enclave loader. The enclave
+Enclave Configuration Data and pass the data to the enclave loader. The code snippet below
+demonstrates an example of how a developer can provide configuration data and pass it to an enclave.
+
+```C
+    const uint8_t config_id_data[64] = {
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,  12,  13,  143, 153,
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,  12,  134, 14,  154,
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,  125, 13,  14,  155,
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 116, 12,  13,  14,  156};
+
+    const uint16_t config_svn = 65535;
+    oe_sgx_enclave_setting_config_data mandatory_config_data_setting = {
+        {0}, config_svn, false /* ignore_if_unsupported */};
```

Comment:
Can you also add an example where the ignore_if_unsupported is set to true? It's an important use case to write an enclave that works for both SGX1 and SGX2, which deserves a paragraph.